client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -756,3 +756,178 @@ export function* trashFile(system, path, id) {
+          return app;
+        }
+        return latest;
+      },
+      { revision: null, version: null }
+    );
+  return latestApp.id;
+};
+
+const getExtractParams = (file, latestExtract) => {
+  const inputFile = `agave://${file.system}${file.path}`;
+  const archivePath = `agave://${file.system}${file.path.substring(
+    0,
+    file.path.lastIndexOf('/') + 1
+  )}`;
+  return JSON.stringify({
+    allocation: 'FORK',
+    appId: latestExtract,
+    archive: true,
+    archivePath,
+    inputs: {
+      inputFile
+    },
+    maxRunTime: '02:00:00',
+    name: 'Extracting Compressed File',
+    parameters: {}
+  });
+};
+
+export function* extractFiles(action) {
+  const extractErrorAction = {
+    type: 'DATA_FILES_SET_OPERATION_STATUS',
+    payload: { status: 'ERROR', operation: 'extract' }
+  };
+  try {
+    const latestExtract = yield call(getLatestApp, 'extract-frontera');
+    const params = getExtractParams(action.payload.file, latestExtract);
+    yield put({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'RUNNING', operation: 'extract' }
+    });
+    const submission = yield call(jobHelper, params);
+    if (submission.execSys) {
+      // If the execution system requires pushing keys, then
+      // bring up the modal and retry the extract action
+      yield put({
+        type: 'SYSTEMS_TOGGLE_MODAL',
+        payload: {
+          operation: 'pushKeys',
+          props: {
+            onSuccess: action,
+            system: submission.execSys,
+            onCancel: extractErrorAction
+          }
+        }
+      });
+    } else if (submission.status === 'ACCEPTED') {
+      yield put({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: 'SUCCESS', operation: 'extract' }
+      });
+    } else {
+      throw new Error('Unable to extract files');
+    }
+  } catch (error) {
+    yield put(extractErrorAction);
+  }
+}
+export function* watchExtract() {
+  yield takeLeading('DATA_FILES_EXTRACT', extractFiles);
+}
+
+/**
+ * Create JSON string of job params
+ * @async
+ * @param {Array<Object>} files
+ * @param {String} zipfileName
+ * @returns {String}
+ */
+const getCompressParams = (files, zipfileName, latestZippy) => {
+  const inputs = {
+    inputFiles: files.map(file => `agave://${file.system}${file.path}`)
+  };
+  const parameters = {
+    filenames: files.reduce((names, file) => `${names}"${file.name}" `, ''),
+    zipfileName
+  };
+  const archivePath = `agave://${files[0].system}${files[0].path.substring(
+    0,
+    files[0].path.lastIndexOf('/') + 1
+  )}`;
+
+  return JSON.stringify({
+    allocation: 'FORK',
+    appId: latestZippy,
+    archive: true,
+    archivePath,
+    maxRunTime: '02:00:00',
+    name: 'Compressing Files',
+    inputs,
+    parameters
+  });
+};
+
+export function* compressFiles(action) {
+  const compressErrorAction = {
+    type: 'DATA_FILES_SET_OPERATION_STATUS',
+    payload: { status: 'ERROR', operation: 'compress' }
+  };
```

Comment:
Why does the status start off as `ERROR`?

Reply:
Ah, I see - this is an action to be used for later if there's an error. Maybe plot this down in that `if` statement below line 900?